signac/core/caching.py

```diff
@@ -0,0 +1,21 @@
+import uuid
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def get_cache():
```

Comment:
@bdice  I have added docstring.
>Are redis stores specific to the application / caller? Do we need to worry about conflicts with other redis clients?

Redis stores are not specific to the application. Redis stores are specific to the port they listen to. We use default port in implementation. So all calls to `redis.Redis()` will connect to the same port. I think we need to handle the conflict with other client.

>I see we're calling this function on every Collection initialization. That's potentially very expensive. Is it possible to create one cache at import time and share it among Collection instances? Is it expensive to call redis.Redis()?

In current implementation, `cache` is class attribute. `redis.Redis()` is called once for a backend.

>If the Redis process is killed while signac is running in a Python shell, what happens? Is there potential for data loss or errors, or does it only slow down future calls to read data?
 
It will raise `redis.exceptions.ConnectionError`. As `cache` is used in buffering so reads and writes operations to buffer will raise error. Should we have a method to switch between redis.client and dict?

Reply:
It sounds like there are unresolved concerns to address here in the Redis backend. **In the interest of getting this PR merged, I would recommend that we disable Redis and make an issue about the remaining items:**
- Regarding (1), we're still missing some important information about Redis, its assumptions, and limits of its functionality.
- Regarding (2), do we need to handle interaction with multiple clients potentially using the same server? Does each signac process need a unique identifier of some kind? I don't know enough about Redis to offer insight here.
- Regarding (4), I'm fine with raising an error but I'd like to test this and get a better understanding for the error and what a user would see if the Redis server fails or is killed. It may be necessary to provide the user some kind of additional error info, I'm not sure. I'd prefer to defer this until a later PR.

I'm fine with the resolution of (3), using this at a class level (so it's not called repeatedly).

Reply:
I'm fine with waiting to add Redis support. Once this PR is done the buffering infrastructure will be in place, so it would be an almost completely independent change. I'm going to leave this code here so that we have it as a base, but I'll change the call to `get_cache` in the `FileBufferedSyncedCollection` to just always make a dict so that this isn't used anywhere for now.